src/compressor/Compressor.h

```diff
@@ -73,11 +73,47 @@ class Compressor {
   QatAccel qat_accel;
 #endif
 
-  static const char* get_comp_alg_name(int a);
-  static boost::optional<CompressionAlgorithm> get_comp_alg_type(const std::string &s);
+  static inline const char* get_comp_alg_name(int a) {
+    auto p = std::find_if(std::cbegin(compression_algorithms), std::cend(compression_algorithms),
+        [a](const auto& kv) { return kv.second == a; });
 
-  static const char *get_comp_mode_name(int m);
-  static boost::optional<CompressionMode> get_comp_mode_type(const std::string &s);
+    if (std::cend(compression_algorithms) == p)
+    return "???"; // It would be nice to revise this...
+
+    return p->first;
+  }
+
+  static inline boost::optional<CompressionAlgorithm> get_comp_alg_type(const std::string &s) {
+
+    auto p = std::find_if(std::cbegin(compression_algorithms), std::cend(compression_algorithms),
+        [&s](const auto& kv) { return kv.first == s; });
+    if (std::cend(compression_algorithms) == p)
+      return {};
+
+    return p->second;
+  }
+
+  static inline const char *get_comp_mode_name(int m) {
+    switch (m) {
+      case COMP_NONE: return "none";
+      case COMP_PASSIVE: return "passive";
+      case COMP_AGGRESSIVE: return "aggressive";
+      case COMP_FORCE: return "force";
+      default: return "???";
+    }
+  }
+
+  static inline boost::optional<CompressionMode> get_comp_mode_type(const std::string &s) {
+    if (s == "force")
+      return COMP_FORCE;
+    if (s == "aggressive")
+      return COMP_AGGRESSIVE;
+    if (s == "passive")
+      return COMP_PASSIVE;
+    if (s == "none")
+      return COMP_NONE;
+    return boost::optional<CompressionMode>();
```

Comment:
shall we consider an unknown compression mode a fatal error?

Reply:
I think we should, but I'm worried about backward compatibility. Any ideas on how to tackle this?

Reply:
On second thought, I think we should leave it to a separate PR